lesson16/src/js/main.js

```diff
@@ -0,0 +1,205 @@
+		categoryButton.textContent = target.category;
+		categoryButton.setAttribute('id','tab-' + (i+1));
+		categoryButton.setAttribute('class', 'tab-button js-tabButton');
+		categoryButton.setAttribute('aria-controls','panel-' + (i+1));
+		categoryButton.setAttribute('role', 'tab');
+		categoryButton.setAttribute('aria-selected', 'false');
+		categoryButton.setAttribute('tabindex', '-1');
+		fragment.appendChild(categoryTab).appendChild(categoryButton);
+	});
+	const categoryTabLists = document.querySelector('.tab-list');
+	categoryTabLists.appendChild(fragment);
+}
+
+const renderNewsArticle = (data) => {
+	const newsLists = document.createElement('ul');
+	newsLists.classList.add('news-list');
+	const fragment = document.createDocumentFragment();
+	const displayedResults = 5;
+	for (let i = 0; i < displayedResults; ++i){
+		const li = document.createElement('li');
+		li.classList.add('news-item');
+
+		const anchor = document.createElement('a');
+		anchor.classList.add('news-link');
+		anchor.href = data[i].link;
+		anchor.textContent = data[i].title;
+		fragment.appendChild(li).appendChild(anchor);
+		addComment(data[i], li);
+		addNewIcon(data[i],li);
+	}
+	newsLists.appendChild(fragment);
+	return newsLists;
+};
+
+function tabContentsInitialDisplay(data){
+	data.forEach((target) =>{
+		const activeCategory = target.isActive;
+
+		if(activeCategory){
+			const categoryButton = categoryTab.querySelector('.js-tabButton');
+			categoryButton.setAttribute('tabindex', '0');
+			categoryButton.setAttribute('aria-selected','true');
+
+			const getSelectedTadID = categoryButton.getAttribute('aria-controls');
+			const selectedPanel = document.getElementById(getSelectedTadID);
+			selectedPanel.setAttribute('aria-hidden', false);
+		}
+	})
+}
+
+function addNewsImage(data){
+	const getNewsArea = document.querySelectorAll('.news-area');
+
+	data.forEach((category,i) => {
+		const imageArea = document.createElement('div');
+		const img = document.createElement('img');
+		img.src = category.image;
+		img.width = 200;
+		img.height = 200;
+		getNewsArea[i].appendChild(imageArea).appendChild(img);
+	})
+
+}
+
+function addComment(data, item) {
+	const commentResult = data.comment;
+	if (commentResult.length) {
+		const commentBox = document.createElement('span');
+		const anchor = document.createElement('a');
+		anchor.classList.add('comment-count');
+		anchor.textContent = commentResult.length;
+		item.appendChild(commentBox).appendChild(anchor);
+	}
+}
+
+function addNewIcon(date,item) {
+	const articleDate = date.date;
+	const articleDateToToday = differenceInDays(new Date(),parseISO(articleDate));
+	const judgmenDays = 60;
+	if(articleDateToToday <= judgmenDays){
+		const newIcon = document.createElement('span');
+		newIcon.textContent = 'new';
+		newIcon.classList.add('icon-new');
+		const newsArticleLink = item.querySelector('.news-link');
+		newsArticleLink.insertAdjacentElement('afterend',newIcon);
+	}
+}
```

Comment:
IMO
Here's one too.
```suggestion
function addNewIcon({ date }, item) {
  const articleDate = date;
  const articleDateToToday = differenceInDays(
    new Date(),
    parseISO(articleDate)
  );
  const judgmenDays = 60;
  if (articleDateToToday <= judgmenDays) {
    const newIcon = document.createElement("span");
    newIcon.textContent = "new";
    newIcon.classList.add("icon-new");
    const newsArticleLink = item.querySelector(".news-link");
    newsArticleLink.insertAdjacentElement("afterend", newIcon);
  }
}
```

Reply:
Thanks for the review!
As advised, I made the icon creation part a different function.

[4d1ba91](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/4d1ba917f6cd15c94935da63bc0f5bf3099dcad7)

Reply:
> IMO
Here's one too.
It is recommended that split assignments be used. Also, it is better to assign to variables what you really use!
こちらも分割代入を使ったほうがいいと思いました！ご確認をお願いします。

Reply:
I have fixed this as well.

[03125e6](https://github.com/nyattsu72/hanson-frontend-cource/pull/21/commits/03125e60d024c88d6b1b7e9a5f79cf192f42715b)